### ISSUE TITLE
Script font (Oldania ADF Std) not used for the campaign outro screen on macOS Sonoma

### ISSUE THREAD

**Issue by irydacea** [wesnoth/wesnoth#8488]
### Game and System Information

- **Version:** 1.18 branch HEAD
- **Build info:** Local build with Xcode 15.2
- **OS:** macOS Sonoma (14.3.1)

### Description of the bug

The Wesnoth campaign outro screen (as in the screen with campaign credits shown upon successful completion of a single-player campaign) normally uses the built-in Oldania ADF Std font for the major text elements in it showing the "The End" text and the campaign's name and credits, and I've verified this works properly on Windows 10 at least.

However, on macOS Sonoma, the game falls back to using a default font instead of Oldania ADF Std for reasons I can't entirely discern, at least when run with default settings. This seems specific to the pangocairo CoreText backend, which is the default in the absence of a suitable alternative set in the `PANGOCAIRO_BACKEND` environment variable at launch, and the native option for macOS.

When using the Fontconfig backend, however, the font is correctly used as expected.

**Default settings:**
(No changes to environment before launch)
<img width="1392" alt="Screenshot 2024-03-02 at 01 34 13" src="https://github.com/wesnoth/wesnoth/assets/489895/c504b717-7e95-49bf-96eb-d29b29067372">

**Fontconfig backend:**
(Run with `PANGOCAIRO_BACKEND=fontconfig wesnoth`)
<img width="1392" alt="Screenshot 2024-03-02 at 01 34 34" src="https://github.com/wesnoth/wesnoth/assets/489895/58a3e156-3556-4ece-9691-f0bbccc482c5">

It is worth mentioning that native macOS utilities such as Finder previews and Font Book seem completely unable to understand the Oldania ADF Std font files for any of its variants, and show an error instead. This is despite, as far as I can tell, macOS still having working, non-deprecated support for OpenType fonts.

<img width="372" alt="Screenshot 2024-03-02 at 01 32 05" src="https://github.com/wesnoth/wesnoth/assets/489895/df558c2b-1c72-4304-9e04-5d99d13691a4">

Assuming I'm not the only affected player somehow (since I seem to have attracted an unreproducible font rendering bug on Windows as well) the options here boil down to either:

1. Forcing the Fontconfig backend for pangocairo on macOS builds with this addition to `main()` under a similar conditional section for Windows:
```c++
#ifdef __APPLE__
	setenv("PANGOCAIRO_BACKEND", "fontconfig", 0);
#endif
```
2. Figuring out why macOS doesn't like Oldania ADF Std and finding a different way around it.

### Steps to reproduce the behavior

1. Complete a campaign or use debug mode to teleport to its epilogue scenario.

**Comment by irydacea** [comment]
As an additional comment, whatever issue there is with the OpenType version of Oldania ADF Std does not apply to the WOFF 2.0 version used for the stable release notes:
<img width="1350" alt="Screenshot 2024-03-02 at 01 55 18" src="https://github.com/wesnoth/wesnoth/assets/489895/16a7699d-6381-4c33-a196-d0b571fdeabb">

**Comment by CelticMinstrel** [comment]
Perhaps it was caused by #8171 , then?

**Comment by irydacea** [comment]
I assume you meant if #8171 could **fix** the issue. I just tested applying the patch therein locally, it does not fix the issue in any way whatsoever — Font Book also continues to refuse to read the fonts.

(My confusion stems from the fact that you're asking if an unmerged PR is causing the issue.)

**Comment by CelticMinstrel** [comment]
Right, I meant maybe it could be caused by that embedding permission that said PR mentions. I guess that was poor phrasing, sorry.

**Comment by Vultraz** [comment]
I did confirm that we're using the exact same version of the font in Debian Bookworm (https://packages.debian.org/bookworm/all/fonts-adf-oldania/download). So it shouldn't be some sort of weird quirk with the version we ship...

**Comment by Wedge009** [comment]
Quoting from #9593:

> Preferably, of course, I'd hear from people on different macOS versions testing this before merging, but that might be hoping for too much since nobody else in the dev team seems to actively run dev builds on macOS...

Is it worth putting out an enquiry on the forums to macOS users as to whether or not they see this issue? It's quite possible the majority of players don't even recognise this as a problem if they haven't played Wesnoth on a system where the Oldania font renders correctly.

**Comment by irydacea** [comment]
> Is it worth putting out an enquiry on the forums to macOS users as to whether or not they see this issue?

That would need to be in the release announcement. Very few people will come across it but it's worth a shot.
(Not entirely sure why we are discussing this here rather than in the PR...?)

**Comment by irydacea** [comment]
Reopened to remind myself to backport this and leave the backport PR parked until the conditions I mentioned in #9593 are met.

**Comment by Wedge009** [comment]
> (Not entirely sure why we are discussing this here rather than in the PR...?)

Because it pertained to the issue itself rather than the changes made in the PR.

**Comment by irydacea** [comment]
Closing since both the dev version of the fix and its backport to 1.18 have been merged by now.
